lib/properties.ts

```diff
@@ -132,9 +134,9 @@ type LanguageDef = {
  * Compiler property fetcher
  */
 export class CompilerProps {
-    private languages: Record<string, any>;
-    private propsByLangId: Record<string, PropertyGetter>;
-    private ceProps: any;
+    public readonly languages: Record<string, any>;
+    public readonly propsByLangId: Record<string, PropertyGetter>;
+    public readonly ceProps: any;
```

Comment:
(mmmayyyyybe is this a PropertyGetter now? or out of scope if it doesn't trivially Just Work™)

Reply:
Ah good point yes it is! I think it should be possible to get `CompilerProps.$getInternal` and `CompilerProps.get` typed properly too

Reply:
beware the creep!!! only if it just works, else it's a todo!

Reply:
I've made the types awesome :)
```ts
const a = this.compilerProps(lang, property); // PropertyValue
const b = this.compilerProps<number>(lang, property); // number
const c = this.compilerProps(lang, property, 42); // number
const d = this.compilerProps(lang, property, undefined, (x) => ["foobar"]); // string[]
const e = this.compilerProps(lang, property, 42, (x) => ["foobar"]); // number | string[]
```

Reply:
wow